Assets/Scripts/Player/PlayerController.cs

```diff
@@ -0,0 +1,171 @@
+                _vignette.intensity.value = Indestructibles.PlayerData.IntoxicationLevel;
+            }
+        }
+        void OnTriggerEnter(Collider other)
+        {
+            if (other.gameObject.CompareTag("NPC"))
+            {
+                var npcManager = other.gameObject.GetComponent<NPCManager>();
+                if (npcManager.IsChasing())
+                {
+                    npcManager.StartPunching();  
+                }
+            }
+
+            if (other.gameObject.CompareTag("Hand") && !_invincible)
+            {
+                var npcManager = other.transform.root.gameObject.GetComponent<NPCManager>();
+                if (npcManager.IsChasing() && npcManager.Punching && !Indestructibles.PlayerData.IsKnockedOut)
+                {
+                    npcManager.OnPlayerHit();
+                    // Play a random grunt sound
+                    if (gruntSounds.Length > 0)
+                    {
+                        _audioSource.PlayOneShot(gruntSounds[Random.Range(0,gruntSounds.Length-1)]);
+                    }
+                    
+                    // Player is dead
+                    if (--_healthPoints <= 0)
+                    {
+                        Indestructibles.PlayerData.IsKnockedOut = true;
+                        Indestructibles.PlayerData.IntoxicationLevel = 0.0f;
+                        _animator.SetTrigger(Animator.StringToHash("Knocked Out"));
+                        gameManager.OnPlayerDeath();
+                        ClearEffects();
+                        return;
+                    }
+                    
+                    // Player got hit but is not dead
+                    var direction = (transform.position - npcManager.transform.position ).normalized;
+                    
+                    // Temporarily disable root motion to add a push force
+                    Indestructibles.PlayerAnimator.applyRootMotion = false;
+                    _rigidbody.AddForce(direction * 5.0f, ForceMode.Impulse);
+                    Invoke(nameof(EnableRootMotion), 0.5f);
+
+                    _invincible = true;
+                    Invoke(nameof(DisableInvincible),2.0f);
+                
+                    // Flicker effect
+                    InvokeRepeating(nameof(FlickerModel),0.0f,0.125f);
+                    Invoke(nameof(StopFlicker),2.0f);
+
+                    // Sober up
+                    Indestructibles.PlayerData.IntoxicationLevel -= 0.5f;
+                    if (Indestructibles.PlayerData.IntoxicationLevel < 0.0f)
+                        Indestructibles.PlayerData.IntoxicationLevel = 0.0f;
+                    _animator.SetFloat(IntoxicationHash, Indestructibles.PlayerData.IntoxicationLevel);
+                    if (_vignette != null)
+                    {
+                        _vignette.intensity.value = Indestructibles.PlayerData.IntoxicationLevel;
+                    }
+                }
+                
+            }
```

Comment:
This code also seems to be in the `NPCManager` class. Will this not cause it to occur twice simultaneously?

Reply:
I don't think it is, the OnCollisionEnter in NPCManager is for when the player bumps into the NPCs when they are wandering so they can start the chase, there are not conflicts